ivy/functional/frontends/tensorflow/math.py

```diff
@@ -297,3 +297,23 @@ def zero_fraction(value, name="zero_fraction"):
 
 
 # TODO: Ibeta for Future Release
+
+
+def accumulate_n(inputs, shape=None, tensor_dtype=None, name="accumulate_n"):
+    if tensor_dtype is None:
+        tensor_dtype = ivy.dtype(inputs[0])
+    if shape is None:
+        shape = ivy.shape(inputs[0])
+    # shape and dtype check
+    for i in range(len(inputs)):
+        ivy.assertions.check_equal(
+            ivy.shape(inputs[i]), shape, message="All inputs must have the same shape."
+        )
+        ivy.assertions.check_equal(
+            inputs[i].dtype,
+            tensor_dtype,
+            message="All inputs must have the same dtype.",
+        )
+    tensor_dtype = ivy.as_native_dtype(tensor_dtype)
```

Comment:
Here we would actually want to convert the dtype to an ivy.dtype not a framework native dtype.